src/shared/GlobalBanners/SelfHostedLicenseExpiration/SelfHostedLicenseExpiration.spec.jsx

```diff
@@ -0,0 +1,574 @@
+          })
+        )
+      })
+    )
+
+    return { user }
+  }
+
+  describe('cloud mode', () => {
+    beforeEach(() => {
+      config.IS_SELF_HOSTED = false
+      setup({ isUndefined: true })
+    })
+    afterEach(() => jest.resetAllMocks())
+
+    it('does not render when there is no provider', () => {
+      render(<SelfHostedLicenseExpiration />, { wrapper: wrapper(['']) })
+
+      const bannerHeader = screen.queryByText(/Missing designated admins/)
+      expect(bannerHeader).not.toBeInTheDocument()
+    })
+
+    it('does not render the banner info when a provider is present', () => {
+      render(<SelfHostedLicenseExpiration />, { wrapper: wrapper() })
+
+      const resolveIssue = screen.queryByText(/Resolve issue/)
+      expect(resolveIssue).not.toBeInTheDocument()
+    })
+  })
+
+  describe('self hosted with invalid data params', () => {
+    beforeEach(() => {
+      config.IS_SELF_HOSTED = true
+    })
+    afterEach(() => jest.resetAllMocks())
+
+    it('does not render when there is no provider', () => {
+      setup({
+        seatsUsed: 5,
+        seatsLimit: 10,
+        expirationDate: '2020-05-09T00:00:00',
+      })
+      render(<SelfHostedLicenseExpiration />, { wrapper: wrapper(['']) })
+
+      const resolveIssueButton = screen.queryByText(/Resolve issue/)
+      expect(resolveIssueButton).not.toBeInTheDocument()
+    })
+
+    it('does not render the banner when there is no license expiration date', async () => {
+      setup({
+        seatsUsed: 5,
+        seatsLimit: 10,
+        expirationDate: null,
+      })
+      render(<SelfHostedLicenseExpiration />, { wrapper: wrapper(['']) })
+
+      const resolveIssueButton = screen.queryByText(/Resolve issue/)
+      expect(resolveIssueButton).not.toBeInTheDocument()
+    })
+
+    it('does not render the banner when there are no seats used', async () => {
+      setup({
+        seatsUsed: null,
+        seatsLimit: 10,
+        expirationDate: '2020-05-09T00:00:00',
+      })
+      render(<SelfHostedLicenseExpiration />, { wrapper: wrapper(['']) })
+
+      const resolveIssueButton = screen.queryByText(/Resolve issue/)
+      expect(resolveIssueButton).not.toBeInTheDocument()
+    })
+
+    it('does not render the banner when there are no seat limit', async () => {
+      setup({
+        seatsUsed: 5,
+        seatsLimit: null,
+        expirationDate: '2020-05-09T00:00:00',
+      })
+      render(<SelfHostedLicenseExpiration />, { wrapper: wrapper(['']) })
+
+      const resolveIssueButton = screen.queryByText(/Resolve issue/)
+      expect(resolveIssueButton).not.toBeInTheDocument()
+    })
+  })
+
+  describe('self hosted with correct params', () => {
+    beforeAll(() => {
+      jest.useFakeTimers().setSystemTime(new Date('2023-08-01'))
+    })
+    beforeEach(() => {
+      config.IS_SELF_HOSTED = true
+    })
+    afterEach(() => jest.resetAllMocks())
+
+    it('does not render the banner when it should not be displayed', async () => {
```

Comment:
Can you update this title to be a little more specific as to why it should not be displayed? Just to give a little more context as to what this is testing.